components/base/sensorcontrolled/spin.go

```diff
@@ -5,97 +5,44 @@ import (
 )
 
 const (
-	increment = 0.01 // angle fraction multiplier to check
-	oneTurn   = 360.0
+	increment   = 0.01 // angle fraction multiplier to check
+	oneTurn     = 360.0
+	slowDownAng = 15. // angle from goal for spin to begin breaking
 )
 
-// Spin commands a base to turn about its center at a angular speed and for a specific angle.
+// // Spin commands a base to turn about its center at a angular speed and for a specific angle.
 func (sb *sensorBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
-	sb.stopLoop()
-	if math.Abs(angleDeg) >= 360.0 {
-		sb.setPolling(false)
-		sb.logger.CWarn(ctx, "feedback for spin calls over 360 not supported yet, spinning without sensor")
-		return sb.controlledBase.Spin(ctx, angleDeg, degsPerSec, nil)
-	}
-	ctx, done := sb.opMgr.New(ctx)
-	defer done()
-	// check if a sensor context has been started
-	if sb.sensorLoopDone != nil {
-		sb.sensorLoopDone()
+	// if controls are not configured, use the underlying base's spin
+	if len(sb.conf.ControlParameters) == 0 {
+		sb.logger.CWarn(ctx, "control parameters not configured, switching to default spin")
+		return sb.controlledBase.Spin(ctx, angleDeg, degsPerSec, extra)
 	}
-
-	sb.setPolling(true)
-
-	// starts a goroutine from within wheeled base's runAll function to run motors in the background
-	if err := sb.startRunningMotors(ctx, angleDeg, degsPerSec); err != nil {
-		return err
-	}
-
-	// this will be removed when spin is migrated to using a control loop
-	if angleDeg > 0 {
-		time.Sleep(1000 * time.Millisecond)
-		sb.logger.Error("done sleeping")
-	}
-
-	// start a sensor context for the sensor loop based on the longstanding base
-	// creator context
-	var sensorCtx context.Context
-	sensorCtx, sb.sensorLoopDone = context.WithCancel(context.Background())
-	if err := sb.stopSpinWithSensor(sensorCtx, angleDeg, degsPerSec); err != nil {
-		return err
+	if sb.orientation == nil {
+		sb.logger.CWarn(ctx, "orientation movement sensor not configured, switching to default spin")
+		return sb.controlledBase.Spin(ctx, angleDeg, degsPerSec, extra)
 	}
 
-	// isPolling returns true when a Spin call is in progress, which is not a success condition for our control loop
-	baseStopped := func(ctx context.Context) (bool, error) {
-		polling := sb.isPolling()
-		return !polling, nil
-	}
-
-	if err := sb.opMgr.WaitForSuccess(
-		ctx,
-		yawPollTime,
-		baseStopped,
-	); err != nil {
-		if !errors.Is(err, context.Canceled) {
+	// make sure the control loop is enabled
+	if sb.loop == nil {
+		if err := sb.startControlLoop(); err != nil {
 			return err
 		}
 	}
-	return nil
-}
-
-func (sb *sensorBase) startRunningMotors(ctx context.Context, angleDeg, degsPerSec float64) error {
-	if math.Signbit(angleDeg) != math.Signbit(degsPerSec) {
-		degsPerSec *= -1
-	}
-	return sb.controlledBase.SetVelocity(ctx,
-		r3.Vector{X: 0, Y: 0, Z: 0},
-		r3.Vector{X: 0, Y: 0, Z: degsPerSec}, nil)
-}
+	sb.setPolling(true)
```

Comment:
is this necessary anymore? I don't 100% remember the specifics but it seems like with controls on both you might not need this anymore

Reply:
this PR currently uses `setPolling` as the primary logic to allow other apis to cancel a Spin. I can switch to using only op manager in this PR if that is preferred.